### PR TITLE
Converts Killer Tomatoes to Basic Mobs

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

Just another lightweight PR porting over a simple animal to the basic mob framework with zero additional AI implementation (it's a killer tomato- it spawns into the world to maul you. how much more intricate does it need to be?).
## Why It's Good For The Game

Able to leverage the basic mob's AI much more effectively, as well as a few niceties that are afforded to us as we further break out of the antiquated framework. 
## Changelog
:cl:
refactor: Expect to see Killer Tomatoes be a bit smarter as they now are able to juggle multiple targets, lock onto you, and barrel towards you. They're much the same as before, but just a little bit more intelligent, which is especially scary considering they're a fucking plant.
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/unique_plant_genes.dm [hunk 1]
```diff
@@ -446,7 +446,7 @@
 /// Killer Tomato's transformation gene.
 /datum/plant_gene/trait/mob_transformation/tomato
 	dangerous = TRUE
-	killer_plant = /mob/living/simple_animal/hostile/killertomato
+	killer_plant = /mob/living/basic/killer_tomato
```
Comment (MrMelbert) [inline code/modules/hydroponics/unique_plant_genes.dm:449]: `/datum/plant_gene/trait/mob_transformation/proc/awaken(obj/item/our_plant)` needs to be updated to support this

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by the-orange-cow** [comment]
I believe the current version of killer tomatoes stats(health, damage, speed) are somewhat based off the stats of the plant(endurance, potency, production speed). 
I'm not savvy enough to tell by just looking, but will this still be the case after refactor?

**Comment by MrMelbert** [comment]
> will this still be the case after refactor?

not currently, that's what the requested review above your comment is covering.

File: code/modules/hydroponics/unique_plant_genes.dm (hunk 1 above)
Comment (san7890) [reply code/modules/hydroponics/unique_plant_genes.dm:459]: ah thanks for catching that i didn't realize that was a feature my bad

File: code/modules/hydroponics/unique_plant_genes.dm [hunk 2]
```diff
@@ -439,14 +439,24 @@
 		spawned_simplemob.melee_damage_lower += round(our_seed.potency * mob_melee_multiplier)
 		spawned_simplemob.melee_damage_upper += round(our_seed.potency * mob_melee_multiplier)
 		spawned_simplemob.move_to_delay -= round(our_seed.production * mob_speed_multiplier)
+
+	if(isbasicmob(spawned_mob))
+		var/mob/living/basic/spawned_basicmob = spawned_mob
+		if(spawned_basicmob.basic_mob_flags & HOSTILE_MOB)
```
Comment (MrMelbert) [inline code/modules/hydroponics/unique_plant_genes.dm:445]: Don't bother with this, the `ishostile` check is just old simplemob crust. Just set the vars regardless

Having a flag or something to check "this is a hostile mob" is a good idea but I'd rather it not be half baked (set on only one mob arbitrarily)

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Pull request merged**